Backpack-SwiftUI/GraphicPromo/Classes/BPKGraphicPromo.swift

```diff
@@ -93,52 +116,83 @@ public struct BPKGraphicPromo: View {
     // swiftlint:disable closure_body_length
     @ViewBuilder
     private func contentView() -> some View {
-        VStack(alignment: .leading, spacing: .md) {
-            if verticalAlignment == .bottom {
-                if let sponsor {
-                    sponsorOverlayView(sponsor)
-                    Spacer()
-                } else {
-                    Spacer()
+        Group {
+            if let sponsor {
+                sponsorContentView(sponsor)
+            } else {
+                Spacer()
+                VStack(alignment: .leading, spacing: .md) {
+                    if verticalAlignment == .bottom {
+                        Spacer()
+                    }
+                    
+                    if let kicker {
+                        BPKText(kicker, style: .label1)
+                            .foregroundColor(variant.foregroundColor)
+                            .lineLimit(nil)
+                    }
+                    
+                    BPKText(headline, style: .heading2)
+                        .foregroundColor(variant.foregroundColor)
+                        .lineLimit(nil)
+
+                    if let subheadline {
+                        BPKText(subheadline, style: .heading5)
+                            .lineLimit(nil)
+                            .foregroundColor(variant.foregroundColor)
+                    }
+                    
+                    if verticalAlignment == .top {
+                        Spacer()
+                    }
                 }
             }
-            
-            if let kicker {
-                BPKText(kicker, style: .label1)
-                    .foregroundColor(variant.foregroundColor)
-                    .lineLimit(nil)
-            }
-            
-            BPKText(headline, style: .heading2)
+        }
+    }
+    
+    @ViewBuilder
+    private func sponsorContentView(_ sponsor: Sponsor) -> some View {
+        VStack(alignment: .leading, spacing: .md) {
+            Spacer()
+            BPKText(Text(headline).bold(), style: .heading2)
```

Comment:
Why do we need a bold headline only for sponsored? Did I miss this from the designs?

Reply:
Good point. Indeed, the font weight on the designs is higher than the one in the default design - hence why It would also make sense to have a hard fork for the content view.

Reply:
Update: correction, the current web product does have a higher weight, 900. 
the default `Headline2` has 700

We have a decision here of:
- Consistency with web
vs
- Consistency with the design


<img width="1562" alt="Screenshot 2024-10-04 at 11 16 53" src="https://github.com/user-attachments/assets/32e93487-78a2-485d-b0bf-986833326a42">

Reply:
I'm not sure Web is using the BPK component or how up to date that is, Graviton might be looking at improving that. So let's go with consistency with the provided design, it's not up to us to ensure consistency across platforms.

Reply:
Consistency it is [92cc002](https://github.com/Skyscanner/backpack-ios/pull/2073/commits/92cc002fc3c3b8b947c09a5131ce5a895fa5d14d)